cheatsheets/Clickjacking_Defense_Cheat_Sheet.md

```diff
@@ -311,3 +312,125 @@ Activate [designMode](https://developer.mozilla.org/en-US/docs/Web/API/Document/
+
+### 3. Content Security Policy (CSP) Frame-Ancestors
+
+Use CSP to control which domains can embed your site:
+
+Content-Security-Policy: frame-ancestors 'self' https://trusted.example.com;
+
+#### Limitations:
+
+CSP-based protections only work when enforced properly and do not protect against all forms of Clickjacking.
+
+Attackers can still use overlays or timing-based click manipulation to deceive users.
+
+### 4. Double-Click Confirmation
+
+Implement a double-click confirmation mechanism for critical actions to prevent unintended interactions.
+#### Example:
+
+<button onclick="confirmAction()">Submit</button>
+<script>
+function confirmAction() {
+    if (!this.clickedOnce) {
+        this.clickedOnce = true;
+        alert('Click again to confirm action.');
+    } else {
+        // Proceed with the action
+    }
+}
+</script>
+
+#### Effectiveness:
+
+Adds an extra layer of user verification.
+
+However, attackers could mimic this behavior to trick users into clicking twice unintentionally.
+
+### 5. Visual Feedback Mechanisms
+
+Ensure that users receive immediate visual feedback on interactions to prevent hidden UI manipulation. Techniques include:
+
+Highlighting clicked elements.
+
+Requiring explicit user confirmation through modal dialogs.
+
+Disabling buttons for a short time after the first click to prevent rapid unintended actions.
+
+Implementing progressive disclosure, where critical actions require an additional confirmation step.
+
+## Additional Considerations
+
+Security research has shown that some traditional defenses, such as X-Frame-Options, Content-Security-Policy (frame-ancestors directive), and the SameSite cookie attribute, may not be effective against all advanced Clickjacking attacks. Attackers can use sophisticated techniques such as dynamic iframe injections, CSS manipulations, and JavaScript-based click tracking to circumvent protections.
+
+- **To improve security:**
+
+Combine multiple defenses rather than relying on a single approach.
+
+Implement real-time click behavior analysis to detect rapid or suspicious interactions.
+
+Use server-side logging to track click patterns and identify potential clickjacking attempts.
+
+Educate users about deceptive UI elements and how to recognize malicious behavior.
+
+## Conclusion
+
+Double Clickjacking is a sophisticated attack that leverages multiple user interactions to bypass traditional Clickjacking defenses. Implementing a combination of X-Frame-Options, Content-Security-Policy, JavaScript-based frame-busting techniques, UI feedback mechanisms, and real-time user behavior analysis can help mitigate this risk effectively. Organizations should adopt a layered security approach to protect against evolving threats and continuously assess the effectiveness of their Clickjacking defenses.
+
+### For a more in-depth understanding of double-click jacking and its implications, you can refer to the following articles:
```

Comment:
I find it curious that all 3 of these referenced in-depth links start out by giving direct credit to Paulos Yibelo who is the one who first discussed (and presumably discovered) Double Clickjacking, and yet you don't link to Yibelo's blog post, https://www.paulosyibelo.com/2024/12/doubleclickjacking-what.html?m=1. That should be the primary reference. I think it's inexcusable to do otherwise.